### PR TITLE
OBGM-386 Move recall logic to service layer

### PR DESCRIPTION
**Pull request by awalkowiak**
To fix issue with not having data from not yet finished session while refreshing product availability for all locations after recall or revert recall

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/inventory/InventoryItemController.groovy [hunk 1]
```diff
@@ -32,7 +31,6 @@ import grails.core.GrailsApplication
 import java.text.DateFormat
 import java.text.SimpleDateFormat
 
-@Transactional
```
Comment (kchelstowski) [inline grails-app/controllers/org/pih/warehouse/inventory/InventoryItemController.groovy:35]: On one hand there is probably no other way to make this work, but on the other hand I'm afraid of that this change considering the other methods, especially the saving ones, that do not use the flush, won't work.

With the setting `allow_update_outside_transaction` we will only auto-commit the change (which is also not good) if we use the flush (this is probably why you've added `flush: true` to the create method).

I would also be aware of the `InventoryItem` `publishPersistenceEvent` - without transaction and with auto-commit (`flush: true` on update) we might face some weird exceptions when the `afterUpdate` would be executed. Have you made sure those work? 

I think the best idea (and the "cleanest" one) would be just to live with the exceptions that might be thrown, create tickets and move the other methods to the service layer.

**Comment by awalkowiak** [comment]
@kchelstowski @jmiranda this one can be re-reviewed

**Review by kchelstowski — APPROVED** [review]

File: grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy [hunk 2]
```diff
@@ -3425,4 +3425,20 @@ class InventoryService implements ApplicationContextAware {
             }
         }
     }
+
+    void recallInventoryItem(InventoryItem inventoryItem) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy:3429]: Given how massive the InventoryService class is we should be thinking longer term about how to refactor this into smaller classes. 

I think the obvious options are 
* Domain (Product, Inventory, InventoryItem) which we used and caused this mess
* Functional area (transfer vs movement vs recall) which seems way better but might have some overlap that we need to contend with 
* Other suggestions?

**Review by jmiranda — COMMENTED** [review]
Approved. See comment for tech-huddle topic.

**Pull request merged**